### PR TITLE
Refactor a bunch of go packages to use go/build.

### PR DESCRIPTION
**Pull request by dlorenc**
Fixes:

Related:

### CODE REVIEW

**Comment by dlorenc** [comment]
I think there are some tradeoffs with this approach, but overall I think it's probably correct long term. The risk of moving go-based builds to a standard pipeline is that we'll eventually "drift" from the exact build invocations from upstream projects, but the benefit (as I see it), is that we can more easily roll out changes to all go-based builds.

These could come in the form of new flags passed, or extra features like a go module proxy, automatic dependency sbom generation, etc...

File: cosign.yaml [hunk 1]
```diff
@@ -15,17 +15,20 @@ environment:
...
 
-  - runs: |
-      CGO_ENABLED=0 go build \
-        -trimpath -ldflags "-buildid= -X sigs.k8s.io/release-utils/version.gitVersion=${{package.version}}" \
-        -o "${{targets.destdir}}/usr/bin/cosign" ./cmd/cosign
+  - uses: go/build
+    with:
+      packages: ./cmd/cosign
+      output: cosign
+      ldflags: -X sigs.k8s.io/release-utils/version.gitVersion=${{package.version}}
```
Comment (imjasonh) [inline cosign.yaml:31]: Just checking does variable expansion work as we expect here? I don't think cosign would complain if it was told its git version was the string `${{package.version}}`.

Comment (dlorenc) [reply cosign.yaml:31]: Yep, I tested that!

**Comment by dlorenc** [comment]
> > These could come in the form of new flags passed, or extra features like a go module proxy, automatic dependency sbom generation, etc...
> 
> ...possibly shared build cache, remote hermetic builds, etc...
> 
> I love it. ❤️

If you loved it then you would have put an LGTM on it!

image

**Review by jonjohnsonjr — APPROVED** [review]

**Pull request merged**